src/test/kotlin/uk/gov/dluhc/emsintegrationapi/database/repository/ApprovedPostalVoteApplicationRepositoryTest.kt

```diff
@@ -0,0 +1,25 @@
+package uk.gov.dluhc.emsintegrationapi.database.repository
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.dluhc.emsintegrationapi.testsupport.testdata.buildApprovedPostalApplication
+
+class ApprovedPostalVoteApplicationRepositoryTest : AbstractRepositoryTest() {
+    @Autowired
+    private lateinit var approvedPostalVoteApplicationRepository: ApprovedPostalVoteApplicationRepository
+    @Test
+    fun `should save an approved postal vote application`() {
+        // Given
+        val approvedPostalVoteApplication = buildApprovedPostalApplication()
+
+        // When
+        approvedPostalVoteApplicationRepository.saveAndFlush(approvedPostalVoteApplication)
+
+        // Then
+        val savedApplication =
+            approvedPostalVoteApplicationRepository.findById(approvedPostalVoteApplication.applicationId).get()
+
+        assertThat(savedApplication).usingRecursiveComparison().isEqualTo(approvedPostalVoteApplication)
+    }
```

Comment:
Is it worth adding some non happy path tests e.g.  for testing the constraint  validation is working?

Reply:
yeh suggest doing that as unit test. See OE as example